src/NuGet.Core/NuGet.Configuration/Settings/Settings.cs

```diff
@@ -153,7 +153,8 @@ public string ConfigFilePath
                     validSettingFiles.AddRange(
                         GetSettingsFileNames(root)
                             .Select(f => ReadSettings(root, f))
-                            .Where(f => f != null));
+                            .Where(f => f != null 
+                            && !f.ConfigFilePath.Equals(Path.GetFullPath(Path.Combine(root, configFileName ?? "")), StringComparison.OrdinalIgnoreCase)));
```

Comment:
I'm sure there is a simpler way to do this., you also want to have clear comments to why are you doing what you are doing.